### PR TITLE
Allow generation of Identifier field in Playground if no validation needed

### PR DESCRIPTION
**Pull request by samjsmith**

### CODE REVIEW

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 1]
```diff
@@ -174,6 +175,7 @@ describe('ResourceComponent', () => {
     describe('#generateResource', () => {
         let mockClassDeclaration;
         let mockModelFile;
+        let mockValidator;
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: Not used.

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 2]
```diff
@@ -182,11 +184,15 @@ describe('ResourceComponent', () => {
...
             mockClassDeclaration.getIdentifierFieldName.returns('resourceId');
-
         });
 
-        it('should generate a valid resource with undefined', () => {
-
+        it('should generate a valid resource with an empty ID', () => {
+            let mockField = {
+                getValidator: () => {
+                    return true;
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: Should return a String?

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 3]
```diff
@@ -182,11 +184,15 @@ describe('ResourceComponent', () => {
...
+                }
+            };
+            mockClassDeclaration.getOwnProperty.returns(mockField);
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: You know the identifier field name (in the `beforeEach`) so might be worth pulling that out to a variable and using `mockClassDeclaration.getOwnProperty.calledWith(idFieldName).returns(mockField);`. This is saying that all field names exist and return the same mock, but maybe that's OK.

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 4]
```diff
@@ -289,7 +295,51 @@ describe('ResourceComponent', () => {
 
         });
 
+        it('should generate a valid resource with a random 4-digit ID', () => {
+            let mockField = {
+                getValidator: () => {
+                    return false;
+                }
+            };
+            mockClassDeclaration.getOwnProperty.returns(mockField);
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: Same comment as above.

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 5]
```diff
@@ -289,7 +295,51 @@ describe('ResourceComponent', () => {
...
+            component.onDefinitionChanged.should.be.calledOn;
+        });
+
         it('should set definitionError on serializer fail', () => {
+            let mockField = {
+                getValidator: () => {
+                    return false;
+                }
+            };
+            mockClassDeclaration.getOwnProperty.returns(mockField);
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: Same comment as above.

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 6]
```diff
@@ -308,6 +358,13 @@ describe('ResourceComponent', () => {
         });
 
         it('should set definitionError on validation fail', () => {
+            let mockField = {
+                getValidator: () => {
+                    return false;
+                }
+            };
+            mockClassDeclaration.getOwnProperty.returns(mockField);
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: Same comment as above.

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts [hunk 7]
```diff
@@ -289,7 +295,51 @@ describe('ResourceComponent', () => {
 
         });
 
+        it('should generate a valid resource with a random 4-digit ID', () => {
+            let mockField = {
```
Comment (bestbeforetoday) [inline packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: Might be worth creating a Sinon stub instance of `Field` just as a safety net to make sure the function you're stubbing exists -- I think Sinon will complain if you add a `returns()` to a function it hasn't stubbed through introspection. Maybe that's overkill though. Your call.

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 1 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: ok

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 2 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: yep

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 3 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: ok

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 7 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: ok

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 4 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: ok

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 5 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: ok

File: packages/composer-playground/src/app/test/resource/resource.component.spec.ts (hunk 6 above)
Comment (samjsmith) [reply packages/composer-playground/src/app/test/resource/resource.component.spec.ts]: plz re-review

**Comment by nklincoln** [comment]
🐼 no issue? 🐼

**Comment by RyZum** [comment]
Is it possible that this broke the automatic generation of the "$class" variable when submitting a transaction ? Since v 0.14.3 the "$class" variable does not change anymore when I select another transaction via the dropdown menu and I was wondering if this can be the cause. Below is a screenshot using one of the basic examples provided in playground : 
image

**Pull request merged**